state/modelconfig.go

```diff
@@ -137,23 +145,30 @@ func (model *Model) modelConfigValues(modelCfg attrValues) (config.ConfigValues,
 }
 
 // UpdateModelConfigDefaultValues updates the inherited settings used when creating a new model.
-func (model *Model) UpdateModelConfigDefaultValues(attrs map[string]interface{}, removed []string, regionSpec *environs.RegionSpec) error {
+func (m *Model) UpdateModelConfigDefaultValues(
+	attrs map[string]interface{}, removed []string, regionSpec *environs.RegionSpec,
+) error {
 	var key string
 
 	if regionSpec != nil {
 		key = regionSettingsGlobalKey(regionSpec.Cloud, regionSpec.Region)
 	} else {
 		key = controllerInheritedSettingsGlobalKey
 	}
-	settings, err := readSettings(model.st.db(), globalSettingsC, key)
+
+	gen, err := m.ActiveGeneration()
+	if err != nil {
+		return errors.Trace(err)
+	}
+	settings, err := readSettings(m.st.db(), globalSettingsC, key, gen == model.GenerationNext)
 	if err != nil {
 		if !errors.IsNotFound(err) {
-			return errors.Annotatef(err, "model %q", model.UUID())
+			return errors.Annotatef(err, "model %q", m.UUID())
 		}
 		// We haven't created settings for this region yet.
-		_, err := createSettings(model.st.db(), globalSettingsC, key, attrs)
+		_, err := createSettings(m.st.db(), globalSettingsC, key, attrs)
```

Comment:
This looks wrong.
- The key is unchanged and so will attempt to create a current gen settings doc upon not finding a next gen doc.
- We should probably copy the current doc is it exists in this case anyway to ensure additions/removals are properly handled.